src/errors.rs

```diff
@@ -171,57 +171,69 @@ fn remove_component_msg(cs: &Component, manifest: &Manifest, toolchain: &str) ->
 }
 
 fn component_unavailable_msg(cs: &[Component], manifest: &Manifest, toolchain: &str) -> String {
-    assert!(!cs.is_empty());
-
     let mut buf = vec![];
-
-    if cs.len() == 1 {
-        let _ = writeln!(
-            buf,
-            "component {} is unavailable for download for channel '{}'",
-            &cs[0].description(manifest),
-            toolchain,
-        );
-        if toolchain.starts_with("nightly") {
-            let _ = write!(
+    match cs {
+        [] => panic!("`component_unavailable_msg` should not be called with an empty collection of unavailable components"),
```

Comment:
This has added a new panic to code that couldn't panic before. For `rustup` panics are quite a concern, though in an error path is -perhaps- acceptable.

In this case I can see there is only one construction point for the error value that formats with this helper. Please add a doc comment to the variant that the helper will panic; we can look at using https://docs.rs/nonempty/latest/nonempty/ or similar later.

Reply:
@rbtcollins IMHO it didn't. This line corresponds to the original `assert!(!cs.is_empty());` line.

Reply:
... but I'd be happy to change this for the better by, for example, adding a "panic" section in the docs.

Reply:
ah thats fair! I think we should still add a doccomment to the variant, because this is a timebomb otherwise.